File: back/DevC/src/main/java/com/back/devc/global/exception/errorCode/CategoryErrorCode.java
package com.back.devc.global.exception.errorCode;
import com.back.devc.global.exception.ErrorCodeSpec;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
@Getter
@RequiredArgsConstructor
public enum CategoryErrorCode implements ErrorCodeSpec {
    CATEGORY_404_NOT_FOUND(
            HttpStatus.NOT_FOUND,
            "CATEGORY_404_NOT_FOUND",
            "카테고리를 찾을 수 없습니다.");

    private final HttpStatus status;
    private final String code;
    private final String message;
}

File: back/DevC/src/main/kotlin/com/back/devc/global/exception/errorCode/CategoryErrorCode.kt
package com.back.devc.global.exception.errorCode

import com.back.devc.global.exception.ErrorCodeSpec
import org.springframework.http.HttpStatus

enum class CategoryErrorCode(
    override val status: HttpStatus,
    override val code: String,
    override val message: String
) : ErrorCodeSpec {
    CATEGORY_404_NOT_FOUND(
        HttpStatus.NOT_FOUND,
        "CATEGORY_404_NOT_FOUND",
        "카테고리를 찾을 수 없습니다."
    );
}